common/src/main/java/io/netty/util/internal/InternalThreadLocalMap.java

```diff
@@ -164,13 +163,17 @@ public int size() {
     }
 
     public StringBuilder stringBuilder() {
-        StringBuilder builder = stringBuilder;
-        if (builder == null || builder.capacity() > STRING_BUILDER_MAX_CAPACITY) {
-            stringBuilder = builder = new StringBuilder(512);
+        final StringBuilder builder = stringBuilder;
+        if (builder == null) {
+            return stringBuilder = new StringBuilder(512);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract the assignment out of this expression. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)